### ISSUE TITLE
How to name the wildcard param

### ISSUE THREAD

**Issue by jelmerdemaat** [fastify/fastify#5899]
#### You have already researched for similar issues?
Yes

#### What are you trying to achieve, or the steps to reproduce?
Is it possible to name the wildcard param? Right now I have:

```js
fastify.get('/api/*', (request) => {
	const path = request.params['*']

	// request: `/api/one/two/three`
	// path == /one/two/three
})
```
#### What did you expect?
I would like it to be something like:
```js
fastify.get('/api/*:randomName', (request) => {
	const path = request.params.randomName;

	// request: `/api/one/two/three`
	// path == /one/two/three
})
```
#### Context

* *node version*: v22.12.0
* *fastify version*: ^5.1.0
* *os*: Mac

**Comment by mcollina** [comment]
I’ve moved it to Fastify as a feature request.

I think this is _possible_, I just wonder what would be the actual benefit.

**Comment by climba03003** [comment]
I don't think it is an useful feature, it can even be replaced by a hook or anything in the middle.
If you don't like to use array property access, we might default it as "wildcard"? or exist in both?

**Comment by jelmerdemaat** [comment]
Thanks for moving the issue. :) 

It's not that it adds anything super powerful, it's just that I don't like the array access via `params['*']` as I feel like it is more error-prone for developers. So as a developer experience feature, I would like to be able to rename it to be able to do `params.randomName`.

Possibly it could also be easier to write types, for TypeScript users? But I wouldn't know about that.

It's really only about ease of writing. 

In my experience it's a very nice feature in Nuxt for example, where you can [create and name catch-all routes](https://nuxt.com/docs/guide/directory-structure/pages#catch-all-route) using:


```vue:pages/[...slug].vue
// Vue file = pages/[...mySlug].vue

<template>
  <p>{{ $route.params.mySlug }}</p>
</template>
```

**Comment by mahmoodhamdi** [comment]
I think named wildcards would be a nice DX improvement. Coming from Express, the syntax `/*:name` or `:name*` feels natural and makes the code more self-documenting.

As @electrovir mentioned, Express v5 now requires named wildcards, so having support for them in Fastify would also make migration easier for developers coming from Express.

In the meantime, a simple workaround is using a `preHandler` hook to alias it:

```js
fastify.get('/api/*', {
  preHandler: (request, reply, done) => {
    request.params.path = request.params['*']
    done()
  }
}, (request) => {
  const path = request.params.path
})
```

But native support in find-my-way would be cleaner.
